wondrous-app/components/Common/Votes/index.tsx

```diff
@@ -1,94 +1,149 @@
-import RetractVoteIcon from 'components/Icons/retractVote';
 import { useMutation } from '@apollo/client';
-import {
-  UPVOTE_TASK_PROPOSAL,
-  DOWNVOTE_TASK_PROPOSAL,
-  REMOVE_TASK_PROPOSAL_VOTE,
-} from 'graphql/mutations/taskProposal';
-import { ProposalVoteType, PROPOSAL_VOTE_LABELS, STATUS_OPEN } from 'utils/constants';
+import { REMOVE_TASK_PROPOSAL_VOTE, VOTE_FOR_PROPOSAL } from 'graphql/mutations/taskProposal';
+import { STATUS_OPEN } from 'utils/constants';
 import { useMe } from 'components/Auth/withAuth';
+import { VotedForCheckMark, NotVotedCheckmark } from 'components/Icons/VotedForCheckmark';
+import palette from 'theme/palette';
+
 import {
   VoteResultsWrapper,
-  VoteButton,
+  VoterProfilePicture,
+  VoteLabel,
   VoteRowWrapper,
   VoteProgressBar,
   VoteCurrentProgress,
-  VoteRowResult,
-  VotePercentageResult,
-  RetractButton,
-  VoteButtonLabel,
+  VoteRowContentWrapper,
 } from './styles';
 
+const computePercentage = (votes, type) => {
+  const amount = votes?.counts[type] || 0;
+  return Math.round((100 * amount) / votes.totalVotes) || 0;
+};
+
 interface Props {
-  votes: {
-    counts: {
-      approve: number;
-      reject: number;
+  fullScreen: boolean;
+  proposal: {
+    id: string;
+    orgId: string;
+    podId: string;
+    org: {
+      privacyLevel: string;
+    };
+    pod: {
+      privacyLevel: string;
     };
-    userVote: string;
+    votes: {
+      counts: Record<string, number>;
+      userVote: string;
+      totalVotes: number;
+    };
+    voteOptions: string[];
+    voteType: 'binary' | 'custom';
   };
-  fullScreen: boolean;
-  taskId: string;
-  totalVotes: number;
   proposalStatus: string;
+  userInOrg: boolean;
+}
+function VoteOptionsRow({ votes, option, handleVote, handleUndoVote, user, voteType }) {
+  const percentage = computePercentage(votes, option);
+  const userVotedFor = votes?.userVote === option;
+  let optionDisplayName = option;
+  if (voteType === 'binary') {
+    optionDisplayName = option === 'approve' ? 'Yes' : 'No';
+  }
+  return (
+    <VoteRowWrapper key={`${option}`}>
+      <VoteRowContentWrapper onClick={() => handleVote(option)}>
+        <div style={{ display: 'flex', gap: 5 }}>
```

Comment:
should this be its own component?

Reply:
You can make this to a styled component

Reply:
Personally I actually thing this is more clear on what it's doing